card/lib/card/query/sql_statement.rb

```diff
@@ -53,27 +55,63 @@ def fields
       end
 
       def joins query
-        [ join_clause(query),
-          query.subqueries.map { |sq| joins sq }
-        ].flatten * "\n"
+        #binding.pry
+        if query.left_joined?
+          join_on_clause query, query.joins.first
+        else
+          [join_on_clause(query, query.joins),
+           query.subqueries.map { |sq| joins sq }
+          ].flatten * "\n"
+        end
+      end
+
+      def join_on_clause query, ready_joins
+        Array.wrap(ready_joins).map do |join|
+          [join_clause(query,join),
+           'ON',
+           on_clause(query, join)
+          ] * ' '
+        end
+      end
+
+      def join_clause query, join
+        to_table = join.to_table
+        to_table = "(#{to_table.sql})" if Card::Query===to_table
+        table_segment = [to_table, join.to_alias] * ' '
+        if query.left_joined? && join == query.joins.first
+          deeper_joins = [
+            join_on_clause(query, query.joins[1..-1]),
+            query.subqueries.map { |sq| joins sq }
+          ].flatten * "\n"
+          table_segment = "(#{table_segment} #{deeper_joins})"
+        end
+
+        [ join.side, 'JOIN', table_segment].compact * ' '
       end
 
-      def join_clause query
-        query.joins.map do |join|
-          j =  join.to_sql
-          j += " AND #{standard_conditions query}" if join.to_table == 'cards'
-          j
+      def on_clause query, join
+        #binding.pry
+
+        on_conditions = join.conditions
+        on_ids = [
+          "#{join.from_alias}.#{join.from_field}",
+          "#{join.to_alias}.#{join.to_field}"
+          ] * ' = '
```

Comment:
Indent the right bracket the same as the start of the line where the left bracket is.